docs/Debugging.md

```diff
@@ -0,0 +1,85 @@
+Debugging
+=========
+
+SGX-LKL provides a wrapper around gdb called `sgx-lkl-gdb`.
+This wrapper is compatible with IDEs and can be used instead of `gdb`.
+`sgx-lkl-gdb` automatically loads the SGX-LKL gdb plugin which
+ensures that debug symbols (if available) are loaded correctly. 
+When running in hardware mode, `sgx-lkl-gdb` uses the corresponding SGX debug
+instructions to read from and write to enclave memory.
+
+## Prerequisites
+
+### Open Enclave
+
+`sgx-lkl-gdb` relies on the Open Enclave gdb plugin.
+[Install Open Enclave](https://github.com/openenclave/openenclave/blob/master/docs/GettingStartedDocs/install_oe_sdk-Ubuntu_18.04.md) if necessary before continuing.
+
+The instructions below assume that Open Enclave was installed in the default directory `/opt/openenclave`.
+If you used a different directory, run `. <OE_PREFIX>/share/openenclave/openenclaverc` first.
+
+### Build mode
+
+Debugging enclave code (SGX-LKL as well as the application code) is only possible in debug and non-release builds of SGX-LKL.
+
+### Encrypted disks
+
+TODO add note on how to handle situations when disk keys are retrieved from a remote service  (either re-create disks unencrypted, re-create disks encrypted but with debug-enabled key release policy (closest to production), or hard-code key in app config)
+
+### Application type
+
+Debugging of non-native applications like Python, Java or .NET applications is currently not possible.
```

Comment:
I don't understand what that means. What are "non-native applications"?

Reply:
Anything that runs in a language runtime, like Python scripts.

Reply:
Ok, in that case, we should remove the statement that debugging of these applications is not possible. It actually depends how a debugger for a specific runtime works, and there is probably no point in going into that much detail here. (For example, debugging functionality that is part of an interpreter (or uses gdb underneath) may actually.) We should just claim `gdb` support and leave it at that.